src/gui/tray/ActivityItemActions.qml

```diff
@@ -0,0 +1,134 @@
+        model: root.activityLinks.length > root.maxActionButtons ? 1 : root.activityLinks.length
+
+        ActivityActionButton {
+            id: activityActionButton
+
+            readonly property int actionIndex: model.index
+            readonly property bool primary: model.index === 0 && root.activityLinks[actionIndex].verb !== "DELETE"
+
+            Layout.minimumWidth: primary ? 100 : 80
+            Layout.preferredHeight: parent.height
+            Layout.preferredWidth: primary ? -1 : parent.height
+            Layout.fillHeight: true
+
+            text: root.actionButtonText(actionIndex)
+            toolTipText: root.activityLinks[actionIndex].label
+
+            imageSource: root.actionButtonIcon(actionIndex, Style.ncBlue)
+            imageSourceHover: root.actionButtonIcon(actionIndex, Style.ncTextColor)
+
+            textColor: imageSource !== "" ? Style.ncBlue : Style.unifiedSearchResulSublineColor
+            textColorHovered: imageSource !== "" ? Style.lightHover : Style.unifiedSearchResulTitleColor
+
+            bold: primary
+
+            onClicked: root.triggerAction(actionIndex)
+        }
+    }
+
+    Loader {
+        // actions that do not fit maxActionButtons limit, must be put into a context menu
+        id: moreActionsButtonContainer
+
+        Layout.preferredWidth: parent.height
+        Layout.fillHeight: true
+
+        active: root.displayActions && (root.activityLinks.length > root.maxActionButtons)
+
+        sourceComponent: Button {
+            id: moreActionsButton
+
+            anchors.fill: parent
+            anchors.topMargin: 10
+            anchors.bottomMargin: 10
+
+            icon.source: "qrc:///client/theme/more.svg"
+
+            background: Rectangle {
+                color: parent.hovered ? "white" : root.moreActionsButtonColor
+                radius: 25
+            }
+
+            ToolTip.visible: hovered
+            ToolTip.delay: Qt.styleHints.mousePressAndHoldInterval
+            ToolTip.text: qsTr("Show more actions")
+
+            Accessible.role: Accessible.Button
+            Accessible.name: qsTr("Show more actions")
+            Accessible.onPressAction: moreActionsButton.clicked()
+
+            onClicked:  moreActionsButtonContextMenuContainer.open();
+
+            Connections {
+                target: root.flickable
+
+                function onMovementStarted() {
+                    moreActionsButtonContextMenuContainer.close();
+                }
+            }
+
+            ActivityItemContextMenu {
+                id: moreActionsButtonContextMenuContainer
+
+                visible: opened
+
+                anchors.right: moreActionsButton.right
+                anchors.top: moreActionsButton.top
+
+                maxActionButtons: root.maxActionButtons
+                activityItemLinks: root.activityLinks
+
+                onMenuEntryTriggered: function(entryIndex) {
+                    root.triggerAction(entryIndex)
+                }
```

Comment:
```suggestion
                onMenuEntryTriggered: root.triggerAction(entryIndex)
```

Reply:
@claucambra This didn't work when I've tried before. Now, I've tried again and the parameter `entryIndex` seems to be out of scope of this component and I am getting this error when calling it like that:
`qrc:/qml/src/gui/tray/ActivityItemActions.qml:122: ReferenceError: entryIndex is not defined`